compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmPackageGen.java

```diff
@@ -807,17 +813,32 @@ CompiledJarFile generate(BIRPackage module, boolean isEntry) {
         return new CompiledJarFile(getModuleLevelClassName(module.packageID, MODULE_INIT_CLASS_NAME, "."), jarEntries);
     }
 
-    private BIRNode.BIRFunction getMainFunc(List<BIRNode.BIRFunction> funcs) {
-        BIRNode.BIRFunction userMainFunc = null;
-        for (BIRNode.BIRFunction func : funcs) {
-            if (func != null && func.name.value.equals(MAIN_METHOD)) {
-                userMainFunc = func;
-                break;
+    private BIRFunction getMainFunc(BIRPackage module) {
+        BIRFunction userMainFunc = null;
+        if (module.packageID.skipTests) {
+            for (BIRFunction func : module.functions) {
+                if (func != null && func.name.value.equals(MAIN_METHOD)) {
```

Comment:
Can a `func` be null? If not shall we remove the condition?

Reply:
It has been there previously as well. But looks redundant. Removed with https://github.com/ballerina-platform/ballerina-lang/pull/39599/commits/93ee9d88e2b9b7d7a1a2769c11f0f9c605a5343d